app/Core/Controller.php

```diff
@@ -53,6 +53,22 @@ public function __construct(
         $this->tpl = $tpl;
         $this->language = $language;
 
+        /**
+         * This is placed here because the session isn't started yet when
+         * AppSettings::loadSettings() is called.  Plus, this is mostly cosmetic/UI
+         * change and should not apply to things such as date/times stored in the DB.
+         * 
+         * This also doesn't effect JS date-time any, only PHP calls for date-time, as
+         * they use incompatible formats.
+         */
+        if (isset($_SESSION['userdata']) && isset($_SESSION['userdata']['id'])) {
```

Comment:
Because it gets overridden when the config file is loaded, at least it did during my testing.  This allowed the config to still be read and then we update it based on the user settings.

I didn't move this closer to the config parser because that doesn't have access to the user session (for the user ID) nor is the session started yet, so even if I set the DI for SettingsRepo/Service I wouldn't be able to get the proper setting.

Reply:
Yeah, the issue I have with this here is that the Base controller should not have anything to do with session or user management. it's for controller management and so in a few months I will have forgotten that there is anything related to session in there. For the problem you mentioned please move it to the header controller which has some session calls already: https://github.com/Leantime/leantime/blob/master/app/Domain/Pageparts/Controllers/Header.php

Reply:
Ah ok yeah makes sense.  Alright I'll move it over today.